### PR TITLE
import charm

### PR DESCRIPTION
**Pull request by mccanne**
This commit imports the charm package from github.com/mccanne.
This way we'll be able to work on it and extend it as discussed.

We also consolidated the uses golang.org/x/crypto/ssh/terminal
with a wrapper package in pkg/terminal (which is also used by
charm to get the interactive terminal width for text formatting).

### CODE REVIEW

**Review by nwt — APPROVED** [review]

File: pkg/terminal/terminal.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package terminal
+
+import (
+	"os"
+
+	"golang.org/x/crypto/ssh/terminal"
```
Comment (mattnibs) [inline pkg/terminal/terminal.go:6]: This package is deprecated, use `golang.org/x/term`

File: pkg/charm/charm.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+type Spec struct {
+	Name          string
+	Usage         string
+	Short         string
+	Long          string
+	New           Constructor
+	Hidden        bool
+	HiddenFlags   string
+	RedactedFlags string
```
Comment (mattnibs) [inline pkg/charm/charm.go:27]: Perhaps a comment here for what Hidden vs Redacted flags mean? I was/am confused about the difference.

**Review by mattnibs — APPROVED** [review]
Sad to see this leave its home at github.com/mccanne/charm. Would be nice to have this as its own standalone repo at some point, but I get the desire to put it here so we can move faster.

File: pkg/charm/charm.go (hunk 2 above)
Comment (mccanne) [reply pkg/charm/charm.go:27]: Let's move charm in "as is" then improve it.  Frankly I don't remember the difference between hidden and redacted right now :-)

File: pkg/terminal/terminal.go (hunk 1 above)
Comment (mccanne) [reply pkg/terminal/terminal.go:6]: good idea

**Pull request merged**